File: MojPrijevozFront/moj_prijevoz_admin/lib/resources/responses/transaction/transaction_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:moj_prijevoz_admin/common/resources/enums/transaction_side.dart';
import 'package:moj_prijevoz_admin/resources/responses/fare/fare_response.dart';
import 'package:moj_prijevoz_admin/resources/responses/transaction/all_transactions_response.dart';

part 'transaction_response.g.dart';

@JsonSerializable()
class TransactionResponse extends AllTransactionsResponse {
  TransactionResponse({
    required super.id,
    required super.fareId,
    required super.side,
    required super.amount,
    super.postedAt,
    required super.createdAt,
    super.fare,
  });

  factory TransactionResponse.fromJson(Map<String, dynamic> json) =>
      _$TransactionResponseFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$TransactionResponseToJson(this);
}

File: MojPrijevoz/MojPrijevoz.Model/Responses/Transaction/TransactionResponse.cs
using MojPrijevoz.Database;

namespace MojPrijevoz.Model.Responses.Transaction;

public class TransactionResponse
{
    public int Id { get; set; }

    public int FareId { get; set; }

    public TransactionSide Side { get; set; }

    public float Amount { get; set; }

    public DateTime? PostedAt { get; set; }

    public DateTime CreatedAt { get; set; }
}